src/test/java/seedu/linkedout/model/applicant/JobTest.java

```diff
@@ -8,17 +9,78 @@
 
 public class JobTest {
 
+    static final int ASCII_CHARACTER_EXCLAMATION_MARK = 33;
+    static final int ASCII_CHARACTER_BACKSLASH = 47;
+    static final int ASCII_CHARACTER_COLON = 58;
+    static final int ASCII_CHARACTER_AT = 64;
+    static final int ASCII_CHARACTER_SQUARE_BRACKET = 91;
+    static final int ASCII_CHARACTER_BACK_TICK = 96;
+    static final int ASCII_CHARACTER_LEFT_CURLY_BRACE = 123;
+    static final int ASCII_CHARACTER_DEL = 127;
+
     @Test
-    public void constructor_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new Job(null));
+    public void constructor_emptyJob_throwsIllegalArgumentException() {
+        String emptyJob = "";
+        assertThrows(IllegalArgumentException.class, () -> new Job(emptyJob));
     }
 
     @Test
     public void constructor_invalidJob_throwsIllegalArgumentException() {
-        String invalidJob = "";
+        String invalidJob = "#";
         assertThrows(IllegalArgumentException.class, () -> new Job(invalidJob));
     }
 
+    @Test
+    public void constructor_invalidJobWhitespace_throwsIllegalArgumentException() {
+        String invalidJob = " ";
+        assertThrows(IllegalArgumentException.class, () -> new Job(invalidJob));
+    }
+
+    @Test
+    public void constructor_validJob_valueEqualsParameter() {
+        String validJob = "job1";
+        Job job = new Job(validJob);
+        assertEquals(job.value, "job1");
+    }
+
+    @Test
+    public void isValidJob_whitespace_false() {
```

Comment:
I think this is covered by the isValidJob test below, where by we assertFalse for job.isValidJob

Reply:
In my opinion, the isValidJob test should be deleted as it is bad unit test design, and consists of testing multiple scenarios in one test, which should not be the case is ideal unit test design. (Easier debugging)

Reply:
Thanks for explaining! Agreed.